RFC/src/RFC-0340_VNConsensusOverview.md

```diff
@@ -1,6 +1,244 @@
+aggregated signature for the proposal.
+
+At this point, all replicas know that enough other replicas have received the proposal and are in agreement that it is
+valid.
+
+In Tendermint, replicas would now wait for the timeout period to make sure that the proposal wasn't going to be
+superseded before finalising the proposal. But there is an attack described in the [HotStuff] paper that could stall
+progress at this point.
+
+The HotStuff protocol prevents this by having a final round of confirmations with the leader. This prevents the stalling
+attack and _also_ lets replicas finalise the proposal _immediately_ on receipt of the final confirmation from the
+leader. This lets HotStuff proceed at "network" speed, rather than with a heartbeat dictated by the BFT synchronicity
+parameter.
+
+Although there are 4 round trips of communication between replicas and the leader, the number of messages sent are O(n).
+It's also possible to stagger and layer these rounds on top of each other, so that there are always four voting rounds
+happening simultaneously, rather than waiting for one to complete in its entirety before moving onto the next one.
+Further details are given in the [HotStuff] paper.
+
+#### Forks and byzantine failures
+
+The summary of the HotStuff protocol given above describes the "Happy Path", when there are no breakdowns in
+communication, or when the leader is an honest node. In cases where the leader is unavailable, the protocol will time
+out, the current view will be abandoned, and all replicas will move onto the next view.
+
+If a leader is not honest, replicas will reject its proposal, and move onto the next view.
+
+If there is a temporary network partition, the chain may fork (up to a depth of three), but the protocol guarantees
+safety via the voting mechanism, and the chain will reconcile once the partition resolves.
+
+#### Leader selection
+
+[HotStuff] leaves the leader selection algorithm to the application. Usually, a round-robin approach is suggested for
+its simplicity. However, this requires the replicas to _reliably_ self-order themselves before starting with SMR, which
+is a non-trivial exercise in byzantine conditions.
+
+For Tari DAN committees, the following algorithm is proposed:
+1. Every replica knows the [Node ID] of every other replica in the committee.
+2. For a given _view number_, the Node ID with the closest XOR distance to the hash of the _view number_ will be the
+   leader for that view, where the hash function provides a uniformly random value of the same length as the Node ID.
+
+
+#### Quorum Certificate
+
+A Quorum certificate, or QC is proof that a super-majority of replicas have agreed on a given state. In particular, a QC
+consists of
+* The type of QC (depending on the phase in which the HotStuff pipeline the QC was signed),
+* The _view number_ for the QC
+* A reference to the leaf node in the state tree being ratified,
```

Comment:
`leaf node` should be discussed or referenced before use

Reply:
Agreed.

Reply:
Since a leaf node is just the state tuple plus a link to it's parent, I decided to just remove the reference to leaf node and called it node instead.